jsapp/js/router/accessDenied.tsx

```diff
@@ -17,15 +17,51 @@ export interface AccessDeniedProps {
 }
 const AccessDenied = (props: AccessDeniedProps) => {
   let messageText;
+  let headerText;
+  let bodyText;
+  let errorNumber;
+  const loggedIn = t(
+    `Please [contact the support team] if you think it's an error.`
+  );
+  const loggedOut = t(
+    `Please try logging in using the header button or [contact the support team] if you think it's an error.`
+  );
 
-  if (sessionStore.isLoggedIn) {
-    messageText = t(
-      `Please try logging in using the header button or [contact the support team] if you think it's an error.`
-    );
-  } else {
-    messageText = t(
-      `Please [contact the support team] if you think it's an error.`
-    );
+  // Obtaining error message number
+  if(props.errorMessage){
+    errorNumber = parseInt(props.errorMessage.replace( /[^\d].*/, '' ));
+  }
+  else{
+    errorNumber = 404;
+  }
+  
+  // Conditionally rendering error message based on number
+  switch (errorNumber){
+    case 403 || 401:
+      headerText = t(`Access Denied`);
+      bodyText = t(`You don't have access to this page.`);
+      if (sessionStore.isLoggedIn) {
+        messageText = loggedIn;
+      } else {
+        messageText = loggedOut;
+      }
+      break;
+    case 404:
+      headerText = t(`Access Denied`);
```

Comment:
Looking better! It looks like you set `headerText = t('Access Denied');` twice in this block; what if you initialized it at the start to `let headerText = t('Access Denied');` and removed those two lines?